registration/lib/decorators.tsx

```diff
@@ -0,0 +1,29 @@
+import { ApolloProvider } from '@apollo/react-hooks'
+import ApolloClient from 'apollo-boost'
+import fetch from 'node-fetch'
+
+import { withAuthentication } from './auth/auth'
+
+const client = new ApolloClient({
+    fetch: fetch as any,
+})
+
+export function nextConnect({ withAuth = true }) {
```

Comment:
Use `_app` page to add top-level providers for apollo.
Add apollo provider file in `lib`